internal/api/impl/auth/oauth.go

```diff
@@ -267,24 +267,24 @@ func (e *oAuthEndpoint) codeExchangeHandler(ctx echo.Context) error {
 	uInfo, err := e.buildUserInfo(uInfoBody)
 	if err != nil {
 		e.logWithError(err).Error("Failed to collect user infos.")
-		return shared.UnauthorizedError
+		return shared.InternalError
 	}
 
 	// Save the user in database
 	user, err := e.svc.SyncUser(uInfo)
 	if err != nil {
 		e.logWithError(err).Error("Failed to sync user in database.")
-		return shared.UnauthorizedError
+		return shared.InternalError
 	}
 
 	username := user.GetMetadata().GetName()
 	_, err = e.tokenManagement.accessToken(username, ctx.SetCookie)
 	if err != nil {
-		return shared.UnauthorizedError
+		return shared.InternalError
```

Comment:
can you add a log everytime we are doing that ? Just so we have a trace of the actual error